File: doc-examples/r2dbc-example-kotlin/src/main/kotlin/example/AuthorController.kt
package example

import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

@Controller("/authors")
class AuthorController(private val repository: AuthorRepository) {
    @Get
    fun all(): Flux<Author> { // <1>
        return repository.findAll()
    }

    @Get("/id")
    fun get(id: Long): Mono<Author> { // <2>
        return repository.findById(id)
    }
}

File: doc-examples/r2dbc-example-python/src/main/python/example/AuthorController.py
from micronaut.http.annotation import Controller, Get
from reactor.core.publisher import Flux, Mono

from example.Author import Author
from example.AuthorRepository import AuthorRepository


@Controller("/authors")
class AuthorController:

    def __init__(self, repository: AuthorRepository):
        self.repository = repository

    @Get
    def all(self) -> Flux[Author]:  # <1>
        return self.repository.findAll()

    @Get("/id")
    def get(self, id: int) -> Mono[Author]:  # <2>
        return self.repository.findById(id)
